opm/autodiff/SimulatorBase_impl.hpp

```diff
@@ -222,12 +222,13 @@ namespace Opm
             // Increment timer, remember well state.
             ++timer;
             prev_well_state = well_state;
-            // Compute Well potentials (only used to determine default guide rates for group controlled wells)
-            // TODO: add some logic to avoid unnecessary calulations of well potentials.
-            asImpl().computeWellPotentials(wells, state, well_state, well_potentials);
+            // Compute Well potentials if they are needed
+            // Only used to determine default guide rates for group controlled wells
+            if ( param_.getDefault("compute_well_potentials", false ) ) {
```

Comment:
We should avoid querying the parameters every timestep, as that will spam the terminal. Better to store as a bool in the class.